modules/extension/ysld/src/main/java/org/geotools/ysld/parse/Util.java

```diff
@@ -218,7 +218,13 @@ public static Expression color(Object value, Factory factory) {
         if (value != null)
             value = value.toString();
 
-        return color != null ? factory.filter.literal(color) : expression((String) value, factory);
```

Comment:
Rather than changing the behavior within YSLD, you should instead update the GeoTools [color converter factory] (https://github.com/geotools/geotools/blob/master/modules/library/main/src/main/java/org/geotools/util/ColorConverterFactory.java) to properly handle these values.